web/src/components/Header.js

```diff
@@ -6,37 +8,102 @@ import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
 import { push } from 'connected-react-router';
 
-
+import { getIsAdmin } from '../reducers/user';
 import { t } from '../i18n';
 
+import Icon, { faChevronDown, faChevronLeft, faEdit, faSignOutAlt } from './Icons';
+
 class Header extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      ariaExpanded: props.ariaExpanded,
+    };
+    this.handleOutsideClick = this.handleOutsideClick.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener('mousedown', this.handleOutsideClick);
+  }
+
+  handleOutsideClick = e => {
+    if (this.node.contains(e.target)) {
+      return;
+    }
+    this.setState({ ariaExpanded: false })
+  };
+
   handleLogout = e => {
     e.preventDefault();
     const { pushRoute } = this.props;
     pushRoute('/logout');
   };
 
+  toggleDropdown = () => {
+    this.setState(prev => ({ ariaExpanded: !prev.ariaExpanded }));
+  };
+
   render() {
-    const { authenticated } = this.props;
+    const { authenticated, currentUser, isAdmin, showSiteTitle } = this.props;
+    const { ariaExpanded } = this.state;
     return (
-      <header className="header__slim">
+      <header ref={node => this.node = node}>
         <div className="ds-l-container">
           <div className="ds-l-row">
-            <div className="ds-l-col--12 ds-l-md-col--4">
-              <Link to="/" className="site-title">
-                {t('titleBasic')}
-              </Link>
+            <div className="ds-l-col--12 ds-l-md-col--4 site-title">
+              {showSiteTitle &&
+                <Link to="/">
+                  {t('titleBasic')}
+                </Link>
+              }
+              {!showSiteTitle && !authenticated &&
+                <Link to="/">
+                  {t('titleBasic')}
+                </Link>
+              }
+              {!showSiteTitle && authenticated && (
+                <Link to="/">
+                  <Icon icon={faChevronLeft} size="sm" />
+                  { isAdmin ? "Admin dashboard" : `${currentUser.state.id.toUpperCase()} APD home` }
+                </Link>
+              )}
             </div>
             {authenticated &&
-              <div className="ds-l-col--12 ds-l-md-col--4 ds-u-margin-left--auto">
-                <Button
-                  size="small"
-                  className="action--logout"
-                  variation="transparent"
-                  onClick={this.handleLogout}
-                >
-                  {t('logout')}
-                </Button>
+              <div className="ds-l-col--12 ds-l-md-col--8">
+                <ul className="nav--dropdown" aria-expanded={ariaExpanded}>
+                  <li>
+                    <button
```

Comment:
Made this a regular button instead of a Button element because testing a click event on a child element is a den of vipers.

Reply:
Is this issue here that there are two `Button`s and it's hard to get Enzyme to give you just the one you want?  Because if so...  yep.  😆